src/Illuminate/Database/Eloquent/TracksPreviousAttributes.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace Illuminate\Database\Eloquent;
+
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Support\Arr;
+
+trait TracksPreviousAttributes
+{
+    /**
+     * The previous model attributes.
+     *
+     * @var array
+     */
+    protected $previous = [];
+
+    /**
+     * Begin tracking the models previous attributes.
+     *
+     * @return void
+     */
+    public static function bootTracksPreviousAttributes()
+    {
+        static::updating(function (Model $model) {
+            $model->syncPrevious();
```

Comment:
Do we need to worry about an `updating` event handler returning `false` to bail out of the update and this getting called, storing the previous attributes, when the update did not actually occur?

Reply:
Not sure if the link should honor highlighted lines, but I meant to highlight lines 1107 to 1109 on the Eloquent\Model file,as you can see in the URL's hash

Reply:
Oh right, we could use the `updated` event. I was thinking that `syncOriginal()` took place prior to the `updated` event firing. Let me update this 👍